### PR TITLE
[Reach] game_engine_settings_definition

### PR DESCRIPTION
**Pull request by matty45**

### CODE REVIEW

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 1]
```diff
@@ -57,13 +85,24 @@ namespace blofeld
...
+
+		{ _field_version_greater, _engine_type_haloreach },
 		{ _field_char_enum, "equipment usage excepting auto turret#false will disable all equipment except auto turret", &player_trait_equipment_usage_enum },
+		
 		{ _field_char_enum, "equipment drop", &player_trait_bool_enum },
 		{ _field_char_enum, "infinite equipment", &player_trait_bool_enum },
+
+		{ _field_version_less_or_equal, _engine_type_haloreach, 3 },
+		{ _field_char_integer, "unknown" },
+		{ _field_char_integer, "unknown" },
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:102]: todos here

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 2]
```diff
@@ -9,6 +9,15 @@ namespace blofeld
 		{ _field_version_greater, _engine_type_haloreach },
 		{ _field_long_flags, "should apply trait", &player_traits_vitality_float_flags },
 
+		{ _field_version_less_or_equal, _engine_type_haloreach, 6 },
+		{ _field_char_enum, "damage resistance", &player_trait_damage_resistance_enum },
+		{ _field_char_enum, "body multiplier", &player_trait_body_multiplier_enum },
+		{ _field_char_enum, "body recharge rate", &player_trait_shield_recharge_rate_enum },
+		{ _field_char_enum, "shield multiplier", &player_trait_shield_multiplier_enum },
+		{ _field_char_enum, "shield recharge rate", &player_trait_shield_recharge_rate_enum },
+		{ _field_char_enum, "shield recharge rate 2", &player_trait_shield_recharge_rate_enum }, // Could be shield stun duration?
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:18]: todo here

File: Tags/TagDefinitions/blofeld/game/game_engine_group.cpp [hunk 3]
```diff
@@ -419,15 +485,27 @@ namespace blofeld
...
-		"teams enabled",
-		"round reset players",
-		"round reset map",
-		"perfection enabled",
-		"mosh",
-		"drop weapons on death",
-		"killcam enabled",
-		"medal scoring enabled",
-		"asymmetric round scoring"
+			{
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game_engine_group.cpp:488]: undo this change

File: Tags/TagDefinitions/blofeld/game/game.cpp [hunk 4]
```diff
@@ -6,10 +6,27 @@ namespace blofeld
...
-		"heroic",
-		"legendary"
+		{
+			_engine_type_haloreach,
+				_versioned_string_list_mode_new,
+			{
+				"unchanged",
+				"east",
+				"normal",
+				"heroic",
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game.cpp:16]: undo this and make a new enum in game_engine_group

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 5]
```diff
@@ -105,25 +164,52 @@ namespace blofeld
...
+
+		{ _field_version_greater, _engine_type_haloreach, 5 },
 		{ _field_char_enum, "vision mode", &player_trait_bool_enum },
 		{ _field_char_enum, "battle awareness", &player_trait_bool_enum },
 		{ _field_char_enum, "threat view", &player_trait_bool_enum },
 		{ _field_char_enum, "aural enhancement", &player_trait_bool_enum },
 		{ _field_char_enum, "nemesis", &player_trait_bool_enum },
+
+		{ _field_version_less_or_equal, _engine_type_haloreach },
+		{ _field_char_integer, "unknown" },
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:211]: add a todo here

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 6]
```diff
@@ -105,25 +164,52 @@ namespace blofeld
 		{ _field_char_enum, "waypoint", &player_trait_waypoint },
 		{ _field_char_enum, "gamertag visible", &player_trait_waypoint },
 		{ _field_char_enum, "aura", &player_trait_aura },
+		
+		{ _field_version_less_or_equal, _engine_type_haloreach },
+		{ _field_char_enum, "forced color", &player_trait_forced_color_enum },
+
+		{ _field_version_less_or_equal, _engine_type_haloreach, 3 },
+		{ _field_char_integer, "unknown" },
+		{ _field_char_integer, "unknown" },
```
Comment (theTwist84) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:173]: todos here

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 7]
```diff
@@ -105,25 +164,52 @@ namespace blofeld
...
 	{
+		{ _field_version_greater, _engine_type_haloreach },
 		{ _field_long_flags, "should apply trait", &player_traits_sensors_float_flags },
+
+		{ _field_version_greater, _engine_type_haloreach, 2 },
 		{ _field_real, "motion tracker range" },
 		{ _field_real, "nemesis duration:seconds" },
-		{ _field_char_enum, "motion tracker", &player_trait_motion_tracker },
+		
+		{ _field_char_enum, "motion tracker mode", &player_trait_motion_tracker },
```
Comment (HaydnTrigg) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:193]: why did you rename this field? also why does it have spacing with no versioning information?

File: Tags/TagDefinitions/blofeld/game/game_engine_group.cpp [hunk 8]
```diff
@@ -642,6 +708,25 @@ namespace blofeld
...
+		"1 Generator",
+		"2 Generators",
+		"3 Generators"
+	};
+	STRING_LIST(game_engine_survival_generator_count_enum, game_engine_survival_generator_count_enum_strings, _countof(game_engine_survival_generator_count_enum_strings));
+
+	STRINGS(game_engine_survival_variant_difficulty_enum)
+	{
+		"unchanged",
+		"east",
```
Comment (theTwist84) [inline Tags/TagDefinitions/blofeld/game/game_engine_group.cpp:723]: Shouldn't this easy not east

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp [hunk 9]
```diff
@@ -105,25 +164,52 @@ namespace blofeld
...
 	};
 
 	TAG_BLOCK(player_traits_sensors_block, 1)
 	{
+		{ _field_version_greater, _engine_type_haloreach },
 		{ _field_long_flags, "should apply trait", &player_traits_sensors_float_flags },
+
+		{ _field_version_greater, _engine_type_haloreach, 2 },
 		{ _field_real, "motion tracker range" },
 		{ _field_real, "nemesis duration:seconds" },
```
Comment (theTwist84) [inline Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:191]: why do you have two sets of `{ _field_version_greater, _engine_type_haloreach },` when only one is required?

File: Tags/TagDefinitions/blofeld/game/game_engine_group.cpp (hunk 8 above)
Comment (matty45) [reply Tags/TagDefinitions/blofeld/game/game_engine_group.cpp:723]: I fixed that in the commits above.

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp (hunk 7 above)
Comment (matty45) [reply Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:193]: i renamed that field as i thought it was not very descriptive about what the field is for.

Comment (matty45) [reply Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:193]: Dont know why the spaces are there, will fix in a later commit

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp (hunk 9 above)
Comment (matty45) [reply Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:191]: Yeah, that does not make sense, ive merged them together now.

File: Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp (hunk 7 above)
Comment (HaydnTrigg) [reply Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:191]: You can't just rename fields on a whim because you think you're more descriptive, this information should be kept the same or versioned out if you want to change it. These names are important

Comment (matty45) [reply Tags/TagDefinitions/blofeld/game/game_engine_player_traits_structures.cpp:191]: allright
